### DISCUSSION TITLE
I made a thread

### DISCUSSION THREAD

**Discussion by Irev-Dev** [zalo/CascadeStudio#48] [Show and tell]
It's a little hacky because it uses a series of lofts, but I don't think extruding along a path is supported atm right?

Alternatively, it's probably possible to do it will a twisted extrude, but it would be very difficult to get the tooth profile right.
The example is up on [CadHub](https://cadhub.xyz/u/irevdev/thread-example)

![thread example](https://user-images.githubusercontent.com/29681384/101980785-0debaf80-3cbc-11eb-9f45-f1cf29e49708.jpg)
![toothProfile](https://user-images.githubusercontent.com/29681384/101980788-104e0980-3cbc-11eb-92ae-cf3678da5f71.jpg)
```javascript

const toothProfile = new Sketch ([0,0]).Fillet(0.1)
    .LineTo([0.4,0.5]).Fillet(0.1)
    .LineTo([0.8,0]).Fillet(0.1)
    .LineTo([0.85,0])
    .LineTo([-0.05,0])
    .End(true).Wire()

hackHelix(toothProfile, {rotations: 5, pitch: 1, diameter: 5})
Cylinder(2.5, 5.05)

function hackHelix(shape, {diameter = 10, pitch = 1.5, rotations = 5, divisions = 360} = {}) {
    const degIncrement = 360/divisions
    const heightIncrement = pitch/divisions
    const circumferance = diameter*Math.PI
    const rad2Deg = num => num*180/Math.PI
    const pitchAngle = rad2Deg(Math.atan(pitch/circumferance))

    const loftWires = Array.from({length: divisions*rotations+1}).map(
        (_, index) => Rotate([0,0,-1], index*degIncrement,
            Translate([0,diameter/2,index*heightIncrement],
                Rotate([0,1,0], 90,Rotate([0,-1,0], pitchAngle, shape))
            )
        )
    )

    return Loft(loftWires)
}
```
[Tweet too](https://twitter.com/IrevDev/status/1337695653535760390)

**Comment by zalo** [comment]
Very nice profile!  I admit that the Rotated Extrude is a bit buggy right now, so this is likely the best way that you could have handled it!

Very creative loft wire construction!
